onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -1132,88 +1132,225 @@ If mask is provided, mask index (that is position of first 0 in mask, or number
-                                    propagateElemTypeFromInputToOutput(ctx, 0, 0);
-                                    ONNX_NAMESPACE::matmulShapeInference(ctx, 0, 2);
-                                  }));
+  ONNX_MS_OPERATOR_SET_SCHEMA(
+      QuantizeWithOrder,
+      1,
+      OpSchema()
+          .SetDoc(R"DOC(Quantize input matrix to specific layout used in cublaslt.)DOC")
+          .Attr("order_input", "cublasLt order of input matrix", AttributeProto::INT)
+          .Attr("order_output", "cublasLt order of output matrix", AttributeProto::INT)
+          .Input(0, "input", "TODO: input tensor of (ROWS, COLS). if less than 2d, will broadcast to (1, X). If 3d, it is treated as (B, ROWS, COS)", "F")
+          .Input(1, "scale_input", "scale of the input", "S")
+          .Output(0, "output", "output tensor", "Q")
+          .TypeConstraint("Q", {"tensor(int8)"}, "Constrain input and output types to int8 tensors.")
+          .TypeConstraint("F", {"tensor(float16)", "tensor(float)"}, "Constrain to float types")
+          .TypeConstraint("S", {"tensor(float)"}, "Constrain Scale to float32 types")
+          .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+            propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto::INT8, 0);
+            if (!hasInputShape(ctx, 0)) return;
+            auto& input_shape = getInputShape(ctx, 0);
+            updateOutputShape(ctx, 0, input_shape);
+          }));
+
+  ONNX_MS_OPERATOR_SET_SCHEMA(
+      DequantizeWithOrder,
+      1,
+      OpSchema()
+          .SetDoc(R"DOC(Dequantize input matrix to specific layout used in cublaslt. attr to specify output type, float16 or float32)DOC")
+          .Attr("order_input", "cublasLt order of input matrix", AttributeProto::INT)
+          .Attr("order_output", "cublasLt order of output matrix", AttributeProto::INT)
+          .Attr("to", "The output data type, only support TensorProto_DataType_FLOAT (1) and TensorProto_DataType_FLOAT16 (10)", AttributeProto::INT)
+          .Input(0, "input", "TODO: input tensor of (ROWS, COLS). if less than 2d, will broadcast to (1, X). If 3d, it is treated as (B, ROWS, COS)", "Q")
+          .Input(1, "scale_input", "scale of the input", "S")
+          .Output(0, "output", "output tensor", "F")
+          .TypeConstraint("Q", {"tensor(int8)"}, "Constrain input and output types to int8 tensors.")
+          .TypeConstraint("F", {"tensor(float16)", "tensor(float)"}, "Constrain to float types")
+          .TypeConstraint("S", {"tensor(float)"}, "Constrain Scale to float32 types")
+          .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+            propagateElemTypeFromAttributeToOutput(ctx, "to", 0);
+            if (!hasInputShape(ctx, 0)) return;
+            auto& input_shape = getInputShape(ctx, 0);
+            updateOutputShape(ctx, 0, input_shape);
```

Comment:
Same comment as in QuantizeWithOrder

Reply:
Yeah, literally they are same. Padding / remove padding is not supported currently.